File: backend/projectverseAPI/projectverseAPI/Models/Collaboration.cs
using projectverseAPI.Interfaces.Marker;

namespace projectverseAPI.Models
{
    public class Collaboration : IAuthorizableByAuthor
    {
        public Guid Id { get; set; }
        public User? Author { get; set; }
        public Guid AuthorId { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public int Difficulty { get; set; }
        public int PeopleInvolved { get; set; }
        public IList<Technology>? Technologies { get; set; }
        public IList<CollaborationPosition>? CollaborationPositions { get; set; }
        public IList<CollaborationApplicant>? CollaborationApplicants { get; set; } = new List<CollaborationApplicant>();
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: frontend/projectverse/src/data/Collaboration.ts
import CollaborationApplicant from "./CollaborationApplicant";
import CollaborationPosition from "./CollaborationPosition";
import Technology from "./Technology";

interface Collaboration{
  id: string,
  //FIX ME XD
  author : {id : string , username : string},
  authorId : number,
  name : string,
  description : string,
  difficulty : number,
  technologies : Technology[],  
  peopleInvolved : number,
  collaborationPositions : CollaborationPosition[],
  collaborationApplicants : CollaborationApplicant[],
  createdAt : Date,
}

export default Collaboration;



